providers/aws/sns/topics.go

```diff
@@ -0,0 +1,110 @@
+package sns
+
+import (
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/mitchellh/mapstructure"
+	"go.uber.org/zap"
+)
+
+type Topic struct {
+	ID                        uint `gorm:"primarykey"`
+	AccountID                 string
+	Region                    string
+	Owner                     *string
+	TopicArn                  *string
+	Policy                    *string
+	DeliveryPolicy            *string
+	DisplayName               *string
+	SubscriptionsConfirmed    *int
+	SubscriptionsDeleted      *int
+	SubscriptionsPending      *int
+	EffectiveDeliveryPolicy   *string
+	FifoTopic                 *bool
+	ContentBasedDeduplication *bool
+}
+
+func (Topic) TableName() string {
+	return "aws_sns_topics"
+}
+
+func GetTopicAttributes(c *Client, TopicArn *string) map[string]*string {
+	params := &sns.GetTopicAttributesInput{
+		TopicArn: TopicArn,
+	}
+	output, _ := c.svc.GetTopicAttributes(params)
```

Comment:
I think it best to check the error here and propagate it.

Reply:
Addressed